### PR TITLE
modal + modalAuth, Themes Provider added, but not used

### PR DESCRIPTION
**Pull request by MAKSsMaK**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,23 +1,36 @@
...
+import { getTheme } from "shared/theme";
 import { useAppSelector } from "./shared/hooks";
 import Header from "./shared/modules/Header/Header";
-import AuthModal from "./shared/modules/Modal/Modal";
 import "./App.scss";
 
+
 const App = () => {
-  const authModalState = useAppSelector((state) => state.authModal);
+    const [themeModa, setThemeMode] = useState<"light" | "dark">("dark")
```
Comment (MAKSsMaK) [inline src/App.tsx]: create enum for this

File: src/shared/modules/AuthModal/AuthModal.module.scss [hunk 2]
```diff
@@ -0,0 +1,48 @@
+ .icon {
+   height: 1rem;
+ }
+
+ .heading {
+   margin-bottom: 1rem;
+ }
+
+
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.module.scss]: delete enter

File: src/shared/modules/AuthModal/AuthModal.module.scss [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ .subTitle {
+   color: #95ACDA;
+ }
+ .altButton {
+   color: #16B2CA;
+   &:hover {
+     cursor: pointer;
+     opacity:.7;
+   }
+
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.module.scss]: delete enter

File: src/shared/modules/AuthModal/AuthModal.module.scss [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+   border-radius: 2rem;
+   color: #fff;
+   margin-bottom: 0.5rem;
+   text-transform: uppercase;
+ }
+
+ .discordButton {
+   background-color: #434de4;
+ }
+ .twitchButton {
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.module.scss]: add enter

File: src/shared/modules/AuthModal/AuthModal.module.scss [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+   background-color: #6e31df;
+ }
+ .facebookButton {
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.module.scss]: add enter

File: src/shared/modules/AuthModal/AuthModal.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import Modal from "shared/components/Modal/Modal";
+import { useAppDispatch,useAppSelector } from "shared/hooks";
+import { setAuthType } from "store/slices/AuthModalSlice";
+import AuthModalOptions from "utils/constants/AuthModalOptions";
+import discord from "../../../assets/discord.svg";
+import facebook from "../../../assets/facebook.svg";
+import twitch from "../../../assets/twitch.svg";
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.tsx]: improve imports

File: src/shared/modules/AuthModal/AuthModal.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import style from "./AuthModal.module.scss";
+
+
+const AuthModal = () => {
+    const dispatch = useAppDispatch();
+    const type = useAppSelector((state) => state.authModal.typeOfModal);
+    const textData = AuthModalOptions[type];
+
+    const changeAuthType = () => {
+        const newValue = type === "signUp" ? "logIn" : "signUp"
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.tsx]: add enum

File: src/shared/modules/AuthModal/AuthModal.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        dispatch(setAuthType({ typeOfModal: newValue }));
+    }
+
+    return (<Modal>
+            <div className={style.heading}>
+                <p className={style.title}>{textData.title}</p>
+                <p className={style.subTitle}>{textData.subTitle}
+                    {" "}<span className={style.altButton} onClick={changeAuthType}>{textData.altText}</span></p>
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.tsx]: what?

File: src/shared/modules/Header/Header.tsx [hunk 9]
```diff
@@ -1,10 +1,11 @@
 import { useState } from "react";
 import { Link } from "react-router-dom";
+import { useAppDispatch } from "shared/hooks";
 import logo from "../../../assets/logo.webp";
-import { togglePopup } from "../../../store/slices/AuthModalSlice";
+import { setAuthType } from "../../../store/slices";
+import { togglePopup } from "../../../store/slices/ModalSlice";
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.tsx]: improve import

File: src/shared/modules/Header/Header.tsx [hunk 10]
```diff
@@ -16,12 +17,18 @@ const Header = () => {
   };
 
   const handleLoginClick = () => {
-    dispatch(togglePopup({ isOpen: true, typeOfModal: "LogIn" }));
+    dispatch(setAuthType({ typeOfModal: "logIn" }));
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.tsx]: use enum

File: src/shared/theme/index.ts [hunk 11]
```diff
@@ -0,0 +1,6 @@
+import darkTheme from "./darkTheme";
+import lightTheme from "./lightTheme";
+
+export const getTheme = (mode: "light" | "dark") => {
```
Comment (MAKSsMaK) [inline src/shared/theme/index.ts]: add enum

File: src/store/slices/ModalSlice.ts [hunk 12]
```diff
@@ -0,0 +1,26 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+interface ModalState {
```
Comment (MAKSsMaK) [inline src/store/slices/ModalSlice.ts]: why? export the interface to a separate file 🤔

File: src/store/slices/ModalSlice.ts [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+const modalSlice = createSlice({
+    name: "modal",
+    initialState,
+    reducers: {
+        togglePopup(state, action: PayloadAction<ModalState>) {
+            return {
+                ...state,
+                isOpen: action.payload.isOpen,
```
Comment (MAKSsMaK) [inline src/store/slices/ModalSlice.ts]: !isOpen

File: src/shared/modules/AuthModal/AuthModal.tsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+import AuthModalOptions from "utils/constants/AuthModalOptions";
+import discord from "../../../assets/discord.svg";
+import facebook from "../../../assets/facebook.svg";
+import twitch from "../../../assets/twitch.svg";
+import style from "./AuthModal.module.scss";
+
+
+const AuthModal = () => {
+    const dispatch = useAppDispatch();
+    const type = useAppSelector((state) => state.authModal.typeOfModal);
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.tsx]: create selectors for this

File: src/shared/modules/AuthModal/AuthModal.tsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
+import Modal from "shared/components/Modal/Modal";
+import { useAppDispatch,useAppSelector } from "shared/hooks";
+import { setAuthType } from "store/slices/AuthModalSlice";
+import AuthModalOptions from "utils/constants/AuthModalOptions";
+import discord from "../../../assets/discord.svg";
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.tsx]: import img like import { ReactComponent as Edit } from 'assets/icons/edit.svg';

File: src/shared/theme/index.ts [hunk 16]
```diff
@@ -0,0 +1,6 @@
...
+    return mode === "light" ? lightTheme : darkTheme;
```
Comment (MAKSsMaK) [inline src/shared/theme/index.ts]: why? ...🤓

File: src/store/slices/AuthModalSlice.ts [hunk 17]
```diff
@@ -1,29 +1,27 @@
+
 import { createSlice, PayloadAction } from "@reduxjs/toolkit";
 
 interface ModalState {
-    isOpen: boolean;
-    typeOfModal?: "LogIn" | "SignUp" | null;
+    typeOfModal: "logIn" | "signUp";
```
Comment (MAKSsMaK) [inline src/store/slices/AuthModalSlice.ts]: use enum

File: src/shared/modules/Header/Header.tsx [hunk 18]
```diff
@@ -16,12 +17,18 @@ const Header = () => {
...
+    openPopup()
   };
   const handleSignUpClick = () => {
-    dispatch(togglePopup({ isOpen: true, typeOfModal: "SignUp" }));
+    dispatch(setAuthType({ typeOfModal: "signUp" }));
+    openPopup()
   };
 
+  const openPopup = () => {
+    dispatch(togglePopup({ isOpen: true }));
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.tsx]: why? ...🤌

File: src/shared/components/Modal/Modal.tsx [hunk 19]
```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { useAppDispatch, useAppSelector } from "shared/hooks/reduxHook";
+import { togglePopup } from "store/slices/ModalSlice";
+import style from "./Modal.module.scss";
+
+
+const Modal: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const dispatch = useAppDispatch();
+  const isOpen = useAppSelector((state) => state.modal.isOpen);
```
Comment (MAKSsMaK) [inline src/shared/components/Modal/Modal.tsx]: use selectors

File: src/shared/components/Modal/Modal.tsx [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+
+  const handleClose = () => {
+    dispatch(togglePopup({ isOpen: false }));
```
Comment (MAKSsMaK) [inline src/shared/components/Modal/Modal.tsx]: why? ...🤌

File: src/shared/modules/AuthModal/AuthModal.module.scss [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+
+ .title {
+   font-size: 3rem;
+   font-weight: 700;
+ }
+
+ .subTitle {
+   color: #95ACDA;
+ }
+ .altButton {
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.module.scss]: add enter

File: src/shared/modules/AuthModal/AuthModal.module.scss [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+   color: #16B2CA;
+   &:hover {
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.module.scss]: add enter

File: src/shared/modules/AuthModal/AuthModal.tsx (hunk 7 above)
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.tsx]: rename variable

File: src/store/slices/AuthModalSlice.ts (hunk 17 above)
Comment (Destwood) [reply src/store/slices/AuthModalSlice.ts]: add enums to AuthModalSlice and its usage

File: src/shared/modules/Header/Header.tsx [hunk 23]
```diff
@@ -1,52 +1,87 @@
...
-import { useAppDispatch } from "../../hooks/redux";
-import style from "./Header.module.scss";
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import logo from 'assets/logo.webp';
+import { useAppDispatch } from 'shared/hooks';
+import { togglePopup } from 'store/slices/ModalSlice';
+import AuthModalOptions from 'utils/constants/AuthModalOptions';
+import Button from '../../components/Button/Button';
+import Input from '../../components/Input/Input';
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.tsx]: import 🤌

File: src/shared/theme/themes.ts [hunk 24]
```diff
@@ -0,0 +1,8 @@
+import { Themes } from '../../utils/enums';
```
Comment (MAKSsMaK) [inline src/shared/theme/themes.ts]: import 🤌

File: src/shared/modules/AuthModal/AuthModal.tsx [hunk 25]
```diff
@@ -0,0 +1,62 @@
...
+import { useAppDispatch, useAppSelector } from 'shared/hooks';
+import { setAuthType } from 'store/slices/AuthModalSlice';
+import AuthModalOptions from 'utils/constants/AuthModalOptions';
+import { Auth } from 'utils/enums/auth';
+import style from './AuthModal.module.scss';
+
+const AuthModal = () => {
+	const dispatch = useAppDispatch();
+	const type = useAppSelector((state) => state.authModal.typeOfModal);
+	console.log(type);
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.tsx]: delete console

File: src/store/slices/AuthModalSlice.ts [hunk 26]
```diff
@@ -1,29 +1,24 @@
...
-               ...state,
-               isOpen: action.payload.isOpen,
-               typeOfModal: action.payload.typeOfModal
-           }
-        },
-    },
+	name: 'modalAuth',
+	initialState,
+	reducers: {
+		setAuthType(state, action: PayloadAction<Auth>) {
```
Comment (MAKSsMaK) [inline src/store/slices/AuthModalSlice.ts:13]: you can also write as setAuthType(state, { payload })

File: src/shared/components/Input/index.tsx [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+import { Theme } from '@mui/material/styles';
+import TextField from '@mui/material/TextField';
+import { SxProps } from '@mui/system';
+import commonInputStyles from './InputStyle';
+
+interface InputProps {
+	value: string;
+	onChange: (value: string) => void;
+	placeholder?: string;
+	type: 'filled' | 'outlined';
```
Comment (MAKSsMaK) [inline src/shared/components/Input/index.tsx]: as I know TextField has no such type typing

interface InputProps {
  value: string;
  onChange: (value: string) => void;
  placeholder?: string;
  type: React.HTMLInputTypeAttribute;
  variant: TextFieldVariants;
}

look interface TextField 🧐

File: src/shared/modules/AuthModal/AuthModal.tsx [hunk 28]
```diff
@@ -0,0 +1,63 @@
+import discord from 'assets/discord.svg';
+import facebook from 'assets/facebook.svg';
+import twitch from 'assets/twitch.svg';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```
Comment (MAKSsMaK) [inline src/shared/modules/AuthModal/AuthModal.tsx]: delete this

File: src/shared/hooks/index.ts [hunk 29]
```diff
@@ -1,2 +1 @@
-export * from "./redux"
-
+export * from '../../store/hooks';
```
Comment (MAKSsMaK) [inline src/shared/hooks/index.ts]: import

**Pull request merged**